src/Time/Formatting.hs

```diff
@@ -8,31 +8,58 @@
 {-# LANGUAGE TypeApplications    #-}
 {-# LANGUAGE TypeOperators       #-}
 
--- | This module introduces function to format time in desired way.
---
--- Here will be some examples after design is approved.
+{- | This module introduces function to format time in desired way.
+
+__Examples__
+
+>>> seriesF @'[DayUnit, HourUnit, MinuteUnit, SecondUnit] (minute 4000)
+"2d18h40m"
+
+>>> seriesF @'[DayUnit, MinuteUnit, SecondUnit] (minute 4000)
+"2d1120m"
+
+>>>  seriesF @'[HourUnit, SecondUnit, MillisecondUnit] (Time @MinuteUnit $ 3 % 2)
+"90s"
+
+-}
 
 module Time.Formatting
        ( Series (..)
+       , unitsF
        ) where
 
-import Time.Rational (KnownRat, Rat, withRuntimeDivRat)
-import Time.Units (KnownUnitSymbol, Time, floorUnit, toUnit)
+import Time.Rational (Rat, withRuntimeDivRat)
+import Time.Units (AllUnits, KnownRatName, Time, floorUnit, toUnit)
 
 -- | Class for time formatting.
 class Series (units :: [Rat]) where
-    seriesF :: forall unit . (KnownUnitSymbol unit, KnownRat unit)
+    seriesF :: forall unit . KnownRatName unit
             => Time unit
             -> String
 
 instance Series ('[] :: [Rat]) where
+    seriesF :: Time someUnit -> String
     seriesF _ = ""
 
-instance (KnownUnitSymbol unit, KnownRat unit, Series units) => Series (unit ': units :: [Rat]) where
-   seriesF :: forall unit' .
-              (KnownUnitSymbol unit', KnownRat unit')
-           => Time unit'
+instance (KnownRatName unit, Series units) => Series (unit ': units :: [Rat]) where
```

Comment:
Okay, next step is to parametrize implementation by `Time` instead of `Rat`, so we can write
```haskell
seriesF @'[Hour]
```
instead of `seriesF @'[HourUnit]`.